common/buildcraft/silicon/tile/TileLaser.java

```diff
@@ -129,11 +130,18 @@ public long getMaxPowerPerTick() {
 
     @Override
     public void update() {
+        ticks++;
         if (world.isRemote) {
+            // set laser render position on client side
+            if (ticks % (5 + world.rand.nextInt(10)) == 0 || targetPos == null) {
```

Comment:
Replace this with a field ````private final SafeTimeTracker clientLaserMoveInterval = new SafeTimeTracker(5, 10);```` and ````if (clientLaserMoveInterval.markTimeIfDelay(world) || targetPos == null) {````. This also means that you can remove the ticks field. (Currently world.rand.nextInt(10) will return a different value every time, so this might take a lot longer than 5 - 15 ticks to change, which is a bit confusing).

Reply:
Should I do the same for the server side targeting to keep things consistent?

Reply:
Yes please :)